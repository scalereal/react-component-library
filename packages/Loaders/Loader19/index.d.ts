import * as React from 'react';
import { WithStyle } from '../../utils';
import { LoaderProps } from './types';
declare const Loader19: React.FC<LoaderProps> & WithStyle;
export default Loader19;
