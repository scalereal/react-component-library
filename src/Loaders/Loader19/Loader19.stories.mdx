import Loader19 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/Loader19"
    component={Loader19}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        innerColor: {
            control: 'color'
        },
        coreColor: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader19 {...args} />
    </div>
);

<Preview>
    <Story name="Loader19" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexWrap: 'wrap' }}>
        <Loader19 size="S" Color="green" innerColor="red" />
        <Loader19 size="S" Color="red" innerColor="green" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexDirection: 'column' }}>
        <Loader19 size="XXS" />
        <Loader19 size="XS" />
        <Loader19 size="S" />
        <Loader19 size="M" />
        <Loader19 size="L" />
        <Loader19 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader19" />
