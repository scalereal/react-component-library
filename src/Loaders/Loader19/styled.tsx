import styled from 'styled-components';
import { LoaderProps } from './types';

const mixin = (size:any,Color:any) => {
    switch (size) {
        case "XXS":
            return `
                height: 15px;
                width: 15px;
                border: 1px solid transparent;
                border-top: 1px solid ${Color || '#FF5722'};
                &:before,&:after{
                    top: 2px;
                    right: 2px;
                    bottom: 2px;
                    left: 2px;
                    border: 1px solid transparent;
                }
           `;
        case "XS":
            return `
                height: 20px;
                width: 20px;
                border: 2px solid transparent;
                border-top: 2px solid ${Color || '#FF5722'};
                &:before,&:after{
                    top: 2px;
                    right: 2px;
                    bottom: 2px;
                    left: 2px;
                    border: 2px solid transparent;
                }
            `;
        case "S":
            return `
            height: 40px;
            width: 40px;
            border: 4px solid transparent;
            border-top: 4px solid ${Color || '#FF5722'};
            &:before,&:after{
                top: 3px;
                right: 3px;
                bottom: 3px;
                left: 3px;
                border: 4px solid transparent;
            }
            `;
        
        case "M":
            return `
            height: 60px;
            width: 60px;
            border: 5px solid transparent;
            border-top: 5px solid ${Color || '#FF5722'};
            &:before,&:after{
                top: 4px;
                right: 4px;
                bottom: 4px;
                left: 4px;
                border: 5px solid transparent;
            }
            `;

        case "L":
            return `
                border: 6px solid transparent;
                height: 80px;
                width: 80px;
                border-top: 6px solid ${Color || '#FF5722'};
                &:before,&:after{
                    top: 5px;
                    right: 5px;
                    bottom: 5px;
                    left: 5px;
                    border: 6px solid transparent;
                }
            `;
        
        case "XL":
            return `
            height: 120px;
            width: 120px;
            border: 6px solid transparent;
            border-top: 6px solid ${Color || '#FF5722'};
            &:before,&:after{
                top: 7px;
                right: 7px;
                bottom: 7px;
                left: 7px;
                border: 6px solid transparent;
            }
            `;
  
    }
 };

export const Loader = styled.div<LoaderProps>`
    ${pr=> mixin(pr.size,pr.Color)}

    border-radius: 50%;
    -webkit-animation: spin 4s linear infinite;
    animation: spin 4s linear infinite;
    position: relative;

    &:before,
    &:after {
        content: "";
        position: absolute;
        border-radius: 50%;
    }

    &:before {
        border-top-color: ${pr=>  pr.innerColor || '#bad375'};
        -webkit-animation: 3s spin linear infinite;
        animation: 3s spin linear infinite;
    }

    &:after {
        border-top-color: ${pr=>  pr.coreColor || '#26a9e0'};
        -webkit-animation: spin 1.5s linear infinite;
        animation: spin 1.5s linear infinite;
    }

    @-webkit-keyframes spin {
        from {
            -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
        }
        to {
            -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
        }
    }

    @keyframes spin {
        from {
            -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
        }
        to {
            -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
        }
    }

`;

export const LoaderWrapper = styled.div<LoaderProps>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    display: ${pr=>pr.visible?'inline-block':'none'};
`;

export const StyledWrapper = styled.div<LoaderProps>`
    -webkit-box-flex: 0;
    -ms-flex: 0 0 25%;
    flex: 0 0 25%;
    border: 1px solid rgba(255, 255, 255, 0.1);
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    margin: 0;
    position: relative;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    overflow: hidden;
`;
