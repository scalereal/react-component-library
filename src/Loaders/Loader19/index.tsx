import * as React from 'react';
import { WithStyle } from '../../utils';
import { Loader, LoaderWrapper, StyledWrapper } from './styled';
import { LoaderProps } from './types';

const Loader19:React.FC<LoaderProps> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <StyledWrapper {...props}>
                <Loader {...props}/>
            </StyledWrapper>
        </LoaderWrapper>
    );
    
}));

Loader19.displayName="Loader19";
Loader19.defaultProps = {
    size:'S',
    visible: true
};

export default Loader19;
