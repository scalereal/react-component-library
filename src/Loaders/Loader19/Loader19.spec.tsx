import { render } from '@testing-library/react';
import * as React from 'react';
import Loader19 from '.';

describe('Loader19 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader19/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader19
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader19 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});