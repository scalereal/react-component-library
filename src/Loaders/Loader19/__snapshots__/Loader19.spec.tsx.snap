// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader19 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ kHrZKX"
      />
    </div>
  </div>
</div>
`;

exports[`Loader19 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ eEUmGd"
      />
    </div>
  </div>
</div>
`;

exports[`Loader19 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ bKNwAR"
      />
    </div>
  </div>
</div>
`;
